[BUG] `json` data required for GET request

**Describe the bug**
`GET` requests cant provide data for `json` property of `validate()`, so it shouldnt be required/validated, but it seems to happen right now. See example below.

**To Reproduce**
consider following endpoint (in `tests/test_plugin_flask.py`). 
This view can handle `GET` request or `POST` request. `json` property is relevant only for `POST` request, however even `GET` request trigger its validation:
```
@app.route("/api/no_response", methods=["GET", "POST"])
@api.validate(
    json=Query,
)
def no_response():
    return {}
```

```
resp = client.get("/api/no_response")
assert resp.status_code == 200
```

**Error message**
```
AssertionError: b'[{"loc":["order"],"msg":"field required","type":"value_error.missing"}]\n
```

--- a/spectree/plugins/flask_plugin.py
+++ b/spectree/plugins/flask_plugin.py
@@ -140,24 +140,28 @@ class FlaskPlugin(BasePlugin):
         req_cookies: werkzeug.datastructures.ImmutableMultiDict
         """
         req_query = get_multidict_items(request.args) or {}
-        if request.mimetype in self.FORM_MIMETYPE:
-            req_json = get_multidict_items(request.form) or {}
-            if request.files:
-                req_json = {
-                    **req_json,
-                    **get_multidict_items(request.files),
-                }
-        else:
-            req_json = request.get_json(silent=True) or {}
         req_headers = dict(iter(request.headers)) or {}
         req_cookies = get_multidict_items(request.cookies) or {}
+        use_json = json and request.method not in ("GET", "DELETE")
 
         request.context = Context(
             query.parse_obj(req_query) if query else None,
-            json.parse_obj(req_json) if json else None,
+            json.parse_obj(self._fill_json(request)) if use_json else None,
             headers.parse_obj(req_headers) if headers else None,
             cookies.parse_obj(req_cookies) if cookies else None,
         )
+
+    def _fill_json(self, request):
+        if request.mimetype not in self.FORM_MIMETYPE:
+            return request.get_json(silent=True) or {}
+
+        req_json = get_multidict_items(request.form) or {}
+        if request.files:
+            req_json = {
+                **req_json,
+                **get_multidict_items(request.files),
+            }
+        return req_json
 
     def validate(
         self,

--- a/spectree/plugins/starlette_plugin.py
+++ b/spectree/plugins/starlette_plugin.py
@@ -56,9 +56,10 @@ class StarlettePlugin(BasePlugin):
             )
 
     async def request_validation(self, request, query, json, headers, cookies):
+        use_json = json and request.method not in ("GET", "DELETE")
         request.context = Context(
             query.parse_obj(request.query_params) if query else None,
-            json.parse_raw(await request.body() or "{}") if json else None,
+            json.parse_raw(await request.body() or "{}") if use_json else None,
             headers.parse_obj(request.headers) if headers else None,
             cookies.parse_obj(request.cookies) if cookies else None,
         )

--- a/tests/test_plugin_starlette.py
+++ b/tests/test_plugin_starlette.py
@@ -320,7 +320,7 @@ def test_validation_error_response_status_code(
     ],
     indirect=["test_client_and_api"],
 )
-def test_flask_doc(test_client_and_api, expected_doc_pages):
+def test_starlette_doc(test_client_and_api, expected_doc_pages):
     client, api = test_client_and_api
 
     resp = client.get("/apidoc/openapi.json")
@@ -329,3 +329,11 @@ def test_flask_doc(test_client_and_api, expected_doc_pages):
     for doc_page in expected_doc_pages:
         resp = client.get(f"/apidoc/{doc_page}")
         assert resp.status_code == 200
+
+
+def test_starlette_no_response(client):
+    resp = client.get("/api/no_response")
+    assert resp.status_code == 200, resp.text
+
+    resp = client.post("/api/no_response", json={"name": "starlette", "limit": 1})
+    assert resp.status_code == 200, resp.text

